### PR TITLE
log4j.xml relocated and excluded from JAR

### PR DESCRIPTION
**Pull request by l-cortese**
`log4j.xml` relocated due to visibility issues and excluded from generated JAR. `pom.xml` formatted according to coding conventions

### CODE REVIEW

**Comment by iluwatar** [comment]
Hi @l-cortese Can you explain a bit how it works?

File: dao/pom.xml [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+    </dependency>
+  </dependencies>
+
+  <build>
+
+    <!-- 
+    log4j.xml file will be copied both in ${project.build.outputDirectory} 
+    and ${project.build.outputDirectory}. Thanks to Sean Patrick Floyd
```
Comment (iluwatar) [inline dao/pom.xml]: Twice the same?

**Comment by l-cortese** [comment]
Ok, there was a typo in the comment at line 29 of pom.xml, it should be 

> log4j.xml file will be copied both in ${project.build.outputDirectory} and ${project.build.**directory**}.

`${project.build.outputDirectory}` is `target/classes`
`${project.build.directory}` is `target`

`log4j.xml` must be copied under `target/classes` for the project to run correctly in Eclipse (nothing new here). `jar:jar` plugin, by default, makes up a JAR archive by copying the tree structure under `target/classes`, so it would include `log4j.xml` too. To **exclude** it from the JAR, I configured the plugin in the pom and "moved" `log4j.xml` to same level dir `target`, so it can be used (and modified if needed) without affecting the JAR.

**Comment by l-cortese** [comment]
Now... if I wanted to correct the comments, should I treat it like a totally new PR?

**Comment by l-cortese** [comment]
ok, just pushed corrections to the master branch in my repo

**Comment by iluwatar** [comment]
Great, thanks for the explanation. So now that the config file is outside the jar it is possible to modify it and run the `dao.jar` with the altered configuration with some command?

**Comment by l-cortese** [comment]
Exactly, you can run `dao-1.8.0-SNAPSHOT.jar` via commandline specifing some options:

    java -Dlog4j.configuration=file:.\log4j.xml -cp .\dao-1.8.0-SNAPSHOT.jar com.iluwatar.dao.App

paying attention to include referenced libraries in the classpath

**Comment by iluwatar** [comment]
Thank you. It works as you explained.

**Pull request merged**